python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -51,6 +51,22 @@ def destroy_engine(self):
         shotgun_globals.unregister_bg_task_manager(self._task_manager)
         self.site_comm.shut_down()
 
+    def set_global_debug(self, state):
+        """
+        Attempts to tell a project subprocess to set the state of
+        the global debug logging setting. This will never raise
+        an exception, but a warning message will be logged if something
+        causes the RPC call to raise.
+
+        :param bool state: The debug to set.
+        """
+        try:
+            self.site_comm.call_no_response("set_global_debug", state)
+        except Exception:
+            # This really can't be a debug log call, because we might have just
+            # toggled debug logging off, in which case the message would not be
+            # logged.
```

Comment:
haha! classic chicken-egg!

Reply:
Is it possible to know in advance if an RPC is available, or is a specific exception raised when the RPC does not exist?

In any case, the warning could suggest why the call failed. We have a pretty good idea here why it would fail. We could recommend to our users to upgrade their project's tk-desktop to `v2.2.12` to enable that feature.